src/containers/sign-on/SignOnProvider.tsx

```diff
@@ -475,6 +476,8 @@ function mapDispatchToProps(dispatch: Dispatch) {
     onSignIn: (email: string, password: string) =>
       dispatch(signOnAction.signIn(email, password)),
     fetchFollowArtists: () => dispatch(signOnAction.fetchAllFollowArtists()),
+    fetchReferrer: (location: Location) =>
```

Comment:
Maybe this action could take in the referrer handle, rather than taking in location and parsing it out in the sagas - since actions are supposed to be serializable, and it looks like location has some functions attached on it